### ISSUE TITLE
Incorrectly tries to install rootless packages on rootful

### ISSUE THREAD

**Issue by ghost** [SailyTeam/Saily#109]
If a repo has the same package with both `iphoneos-arm` (rootful) and `iphoneos-arm64` (rootless) architecture, Saily will try (and fail) to install the rootless version even when you are on a rootful jailbreak and the `iphoneos-arm64` architecture has not been added to dpkg.

**Comment by Lakr233** [comment]
Saily does not recognize architecture, as not designed to. I'm not planing update for this, instead it is recommended to separate them as two packages.

The main reason for that is apt repo isn't setup the right way in the past, and Xina15 is doing his job as a compatibility layer.

If any developer want to do this, they can add filter inside AptRepository decoder.

**Comment by Lakr233** [comment]
> If any developer want to do this, they can add filter inside AptRepository decoder.

@Lessica let's take a note

**Comment by ghost** [comment]
Technically it's already two different packages, just with the same bundle ID. I'd prefer to avoid using a different bundle ID because other packages may depend on some of my packages. So the other workaround would be making a separate repo for rootless, I guess.

I even tried this, but it didn't fix it:
```
Package: xyz.rootless
Version: 0.1.0
Provides: xyz (= 0.1.0)
```

I was testing on palera1n because users were having issues with Saily. I haven't used XinaA15 as I don't have a compatible device, but assuming it adds both `iphoneos-arm` and `iphoneos-arm64` using `dpkg --add-architecture` (which is the proper way to signal support for both rootful and rootless packages), you could just check `/var/lib/dpkg/arch` for a list of architectures supported on the current jailbreak (one per line).

**Comment by Lakr233** [comment]
I have no device to test the build, so I asked @Lessica if he can help. I understand your case, but it is not possible for me to fix and test it. I can provide you a test build, if you still want me to fix it, but that is on your own to test them out.

**Comment by Lakr233** [comment]
Fixed in rootless 3.0
